sda/internal/database/db_functions.go

```diff
@@ -63,6 +63,42 @@ func (dbs *SDAdb) getFileID(corrID string) (string, error) {
 	return fileID, nil
 }
 
+// GetInboxFilePathFromID checks if a file exists in the database for a given user and fileID
+// and that is not yet archived
+func (dbs *SDAdb) GetInboxFilePathFromID(submissionUser, fileID string) (string, error) {
+	var (
+		err      error
+		count    int
+		filePath string
+	)
+
+	for count == 0 || (err != nil && count < RetryTimes) {
+		filePath, err = dbs.getInboxFilePathFromID(submissionUser, fileID)
+		count++
+	}
+
+	return filePath, err
+}
+
+func (dbs *SDAdb) getInboxFilePathFromID(submissionUser, fileID string) (string, error) {
+	dbs.checkAndReconnectIfNeeded()
+	db := dbs.DB
+
+	const getFilePath = "SELECT submission_file_path from sda.files where " +
+		"submission_user= $1 and id = $2 " +
+		"AND EXISTS (SELECT 1 FROM " +
+		"(SELECT event from sda.file_event_log where file_id = $2 order by started_at desc limit 1) " +
+		"as subquery WHERE event = 'uploaded')"
```

Comment:
Reasonable question, especially since we reached different conclusions about this at different time points. The last final discussion was on (in a thread) with the following outcome:
- If a file has been uploaded and no other action taken - regardless if the setting is FEGA, GDI or BP - the only state that file can have is uploaded .
- A registered  file is not fully uploaded and will not be listable in the bucket. You need the upload ID from the S3 system in order to remove them.
- A file file with any status of submitted, ingested, archived, or error  have been interacted with by the submission process and as such removal in other places than the inbox will be needed.

Reply:
(I'm marking this as resolved now, since you have approved :) )